libraries/engage/app-rating/actions/showModal.js

```diff
@@ -0,0 +1,70 @@
+import showModalAction from '@shopgate/pwa-common/actions/modal/showModal';
+import { getPlatform } from '@shopgate/pwa-common/selectors/client';
+import { historyPush } from '@shopgate/pwa-common/actions/router';
+import appConfig from '@shopgate/pwa-common/helpers/config';
+import { increaseRejectionCount, setLastPopupTimestamp } from '../action-creators/popup';
+import { generateReviewLink } from '../helpers';
+import { isMinTimeBetweenPopupsElapsed } from './minTime';
+
+const {
+  appRating: {
+    bundleId: bId,
+    rejectionLink,
+  },
+} = appConfig;
+
+/**
+ * shows the actual modal
+ * @param {Function} resetAction the reset action function
+ * @param {Function} increaseAction the function to increase the appropriate counter
+ * @param {boolean} mustShow if the modal must be shown
+ * @param {boolean} hasRepeats if the counters has repeats
+ * @return {(function(*, *): void)|*}
+ */
+export function showModal(resetAction, increaseAction, mustShow, hasRepeats) {
+  return (dispatch, getState) => {
+    if (!mustShow && hasRepeats) {
+      dispatch(increaseAction());
+    }
+
+    if (!(mustShow && hasRepeats)) {
+      return;
+    }
+
+    if (!isMinTimeBetweenPopupsElapsed()) {
```

Comment:
you changed this function to be an action now. So it returns now another function and therefor it will evaluate **always** to !true. 

There is no reason this function should be an action. should be a helper or maybe a selector

Reply:
an action should always cause some state changes/ trigger subscriptions